### ISSUE TITLE
hidden categories appearing or not

### ISSUE THREAD

**Issue by desb42** [gnosygnu/xowa#423]
In xowa, at the moment all Categories and Hidden Categories are shown at the bottom of the page.
Mediawiki has different behaviour

For enwiki eg en.wikipedia.org/wiki/India%E2%80%93Pakistan_border_skirmishes_(2019) only Categories are shown (although the Hidden ones are in the html code)

For frwiki eg fr.wikipedia.org/wiki/Mykonos only Categories are shown; however some initialisation javascript adds a [+] to allow for expansion of the Hidden Categories

I would propose that, for wikis that do not have an expanding capability, show the Hidden Categories, and where they do don't

Possibly all the wikis would need to be checked; however as a start, I would suggest that Hidden Categories are always shown except for frwiki (and add any other wikis as they are determined)

This does presuppose that the gadget work #420, #422 has been done

**Comment by gnosygnu** [comment]
Yeah, I opted to show as much info as possible, even if it's not an exact copy of MediaWiki.

In this particular case, I hard-coded the css class to always show. See: https://github.com/gnosygnu/xowa/blob/9e5a13f54f478273962b3b23bd3c747cf9640a51/400_xowa/src/gplx/xowa/addons/wikis/ctgs/htmls/pageboxs/doubles/Xoctg_double_grp.java#L59

For the proposal above, I guess the following would need to be done:
* Change ```mw-hidden-cats-user-shown``` to ```mw-hidden-cats-hidden```
* Integrate Gadget support so that frwiki could add the [+]
* Add a user option to force Hidden Categories to always show, always hide, or do whatever the default wiki behavior is

It's not much work, but I think it can wait some time. Let me know if there's anything I missed above. Thanks.

**Comment by desb42** [comment]
That seems to sum it up succinctly

I have implemented the first two step in my codebase (with one hell of a lot of hackery on javascript startup and css injection)
I think the gadget work is the crux here
